hathor/manager.py

```diff
@@ -592,7 +615,17 @@ def _initialize_components_new(self) -> None:
         # XXX: last step before actually starting is updating the last started at timestamps
         self.tx_storage.update_last_started_at(started_at)
         self.state = self.NodeState.READY
-        self.log.info('ready')
+
+        t1 = time.time()
+        tdt = LogDuration(t1 - t0)
+
+        environment_info = self.environment_info.as_dict() if self.environment_info else {}
+
+        vertex_count = self.tx_storage.get_tx_count() + self.tx_storage.get_block_count()
```

Comment:
I wasn't sure if I should use this or `vertex_count = self.tx_storage.get_count_tx_blocks()`. 

I chose this way because it seems these methods use indexes, while the other one doesn't

Reply:
I tested this, and it seems these methods don't always match on their counts.

In a small local DB I have, I was getting 73 from `get_tx_count/get_block_count` vs 79 from `get_count_tx_blocks`.

There is something about this in a [RocksDB FAQ](https://github.com/aayushKumarJarvis/rocks-wiki/blob/master/RocksDB-FAQ.md), and it seems it's actually expected that the result is not 100% accurate:

> Obtaining an accurate number of keys in any LSM databases like RocksDB is a challenging problem as they have duplicate keys and deletion entries (i.e., tombstones) that will require a full compaction in order to get an accurate number of keys. In addition, if the RocksDB database contains merge operators, it will also make the estimated number of keys less accurate.

Reply:
Then I think we should phase-out `get_count_tx_blocks` in favor of the other two, or rather implement `get_count_tx_blocks` based on the other two.

Reply:
@luislhl I feel we should add a test to it. As far as I understood, this test will fail for RocksDB, right? In this case, I feel RocksDB should override `get_count_tx_blocks()` to do exactly `get_tx_count() + get_block_count()`.

Reply:
Yes, it should fail for RocksDB. I'll try to develop such test.

Reply:
I developed the test and a fix proposal in this commit: https://github.com/HathorNetwork/hathor-core/pull/490/commits/6d5f9723ecb642823bd5f0f6f2981241178f6efa

The funny thing is that, before the fix, my test was only failing for the `CacheRocksDBStorageTest`, but not for the `TransactionRocksDBStorageTest`.

I'm not sure why this happened. While testing manually, I was able to reproduce the problem both with and without cache.

So I proposed a fix that will use `get_tx_count() + get_block_count()` both when using cache and when not using it.

@msbrogli @jansegre

Reply:
This change broke one test that is not trivial to fix. So I decided to move it to a new PR: https://github.com/HathorNetwork/hathor-core/pull/496

Reply:
That makes sense, I think it's a good approach.